src/main/java/com/google/api/codegen/transformer/php/PhpModelTypeNameConverter.java

```diff
@@ -119,10 +133,76 @@ public TypeName getTypeNameForElementType(TypeRef type) {
 
   @Override
   public TypeName getTypeName(ProtoElement elem) {
-    if (elem.getFullName().equals("google.protobuf.Empty")) {
-      return typeNameConverter.getTypeName("\\google\\protobuf\\EmptyC");
+    try {
+      return getTypeName(elem, MAX_NESTED_DEPTH);
+    } catch (IllegalStateException e) {
```

Comment:
This is to help with debugging, so that the top level `ProtoElement` object appears in the exception. This information is not available at the exception throw sites (lines 169 and 173)